Aptoide/aptoide-client-v8
# [ASV-1323] Download url can not be empty

**What does this PR do?**
This PR aims at fixing the error on fabric, where an exception would be thrown (by us) reporting a null path to download. To fix this, we added more verifications to the app before starting the download. Before we were only checking the download paths now we also check the obb paths. Also took the chance to refactor that code a little bit. Finally, to track the dimension of this problem of having apps with null paths, we are also sending an event to Fabric, called INVALID_DOWNLOAD_PATH.

**Database changed?**

 No

**Where should the reviewer start?**

- [ ] DownloadFactory.java

**How should this be manually tested?**
Rewrite a getApp request on app for each of the verifications being made and check that the app does not crash. Try empty paths for both apk and obbs, each at a time to see the different events being sent. Also you can test the other verifications, again using rewrite on charles.

**What are the relevant tickets?**

  Tickets related to this pull-request: [ASV-1323](https://aptoide.atlassian.net/browse/ASV-1323)

**Questions:**

   Does this add new dependencies which need to be added? (Eg. new libs, new keys, etc.) 




**Code Review Checklist**
